### ISSUE TITLE
Multiline and returnLabelType="send' not working together?

### ISSUE THREAD

**Issue by pierroo** [facebook/react-native#43910]
### Description

On android, we unfortunately cannot have the returnLabelType="send" when using multiline.
So it makes it impossible to replicate a "snapchat-like" textinput for example, where we can send our message from the keyboard while having the multiline in place as we type.

There was this old issue about the same: https://github.com/facebook/react-native/issues/33193
and for strange reason it was marked as "resolved" by facebook team when it's not?

### Steps to reproduce

1- add prop multiline to textinput
2- add prop returnLabelType='send' to textinput
3- returnlabeltype is ignored

### React Native Version

0.73.5

### Affected Platforms

Runtime - Android

### Output of `npx react-native info`

```text
System:
  OS: Windows 11 10.0.22631
  CPU: (20) x64 12th Gen Intel(R) Core(TM) i7-12700KF
  Memory: 20.26 GB / 31.86 GB
Binaries:
  Node:
    version: 18.18.2
    path: C:\Program Files\nodejs\node.EXE
  Yarn: Not Found
  npm:
    version: 9.8.1
    path: C:\Program Files\nodejs\npm.CMD
  Watchman: Not Found
SDKs:
  Android SDK: Not Found
  Windows SDK: Not Found
IDEs:
  Android Studio: AI-221.6008.13.2211.9514443
  Visual Studio:
    - 17.8.34309.116 (Visual Studio Community 2022)
Languages:
  Java: 11.0.18
  Ruby: Not Found
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.73.5
    wanted: ^0.73.5
  react-native-windows: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found
```

### Stacktrace or Logs

```text
no crash/error
```

### Reproducer

https://github.com/react-native-community/reproducer-react-native

**Comment by jakex7** [comment]
Hi @pierroo,
It appears you meant `returnKeyType` instead of `returnLabelType`, as it isn't documented or present in the codebase. To achieve your desired functionality, you can set the `submitBehavior` prop with either `submit` or `blurAndSubmit` values on the TextInput component. 

![image](https://github.com/facebook/react-native/assets/39670088/d997abda-689a-40a5-98fc-2a754ec6dfdb)

```tsx
<TextInput
  multiline={true}
  returnKeyType="send"
  submitBehavior="submit"
/>
```


This prop isn't documented, but has been introduced in #33653 and available since version v0.71.
```ts
type SubmitBehavior = 'submit' | 'blurAndSubmit' | 'newline'
```

**Comment by pierroo** [comment]
It works perfectly, thank you so much @jakex7 !
